src/compiler/compile/nodes/Binding.ts

```diff
@@ -77,7 +81,7 @@ export default class Binding extends Node {
 			variable[this.expression.node.type === 'MemberExpression' ? 'mutated' : 'reassigned'] = true;
 
 			if (info.expression.type === 'Identifier' && !variable.writable) {
-				component.error(this.expression.node, {
+				component.error(this.expression.node as any, {
```

Comment:
Not sure what the method signature of `error` is and if we can change it, but maybe we could change that signature instead of doing `as any` here?

Reply:
No, I honestly don't know what's going on here either. `component.error` takes something that has `start` and `end` attributes. `node` in `Expression` is an `estree` `Node` and those have a `loc` instead with the position. I checked what object gets passed here in the tests and it's an object that apart from being an `estree` `Node` also has `start` and `end` attributes added to it. I'm not sure where it gets those and if we can assume they always have them.